crates/ty_python_semantic/src/types.rs

```diff
@@ -10861,9 +10866,9 @@ impl<'db> UnionType<'db> {
     pub(crate) fn filter(
         self,
         db: &'db dyn Db,
-        filter_fn: impl FnMut(&&Type<'db>) -> bool,
+        mut f: impl FnMut(&Type<'db>) -> bool,
     ) -> Type<'db> {
-        Self::from_elements(db, self.elements(db).iter().filter(filter_fn))
+        Self::from_elements(db, self.elements(db).iter().filter(|ty| f(ty)))
```

Comment:
Can we make the callback take in `Type<'db>` instead of `&Type<'db>`? I find that much more ergonomic, and I think it just requires a `.copied()` before the `filter`. (Ditto above in `Type::filter`)

Reply:
There are some `Type` methods that take `&self`, so we should probably address them all at once. Only changing `filter` would mean `.filter(Type::is_typed_dict)` no longer works, for example.